src/api/cristinApi.ts

```diff
@@ -111,9 +111,16 @@ export const fetchOrganization = async (id: string) => {
   const fetchOrganizationResponse = await apiRequest2<Organization>({
     url: id,
   });
+  setIdentifierFromId(fetchOrganizationResponse.data);
   return fetchOrganizationResponse.data;
 };
 
+function setIdentifierFromId(org: Organization) {
+  org.identifier = getIdentifierFromId(org.id);
+  org.hasPart?.map((subOrg) => setIdentifierFromId(subOrg));
+  org.partOf?.map((subOrg) => setIdentifierFromId(subOrg));
+}
```

Comment:
Ikkw helt åpenbart hvorfor denne er er her, så kanksje legg til en liten kommentar?

Reply:
dette er bare for å legge til `identifier` for alle objekter, sånn at de enkelt kan brukes i query params det?

Reply:
Vil ikke denne kjøre uendelig? 🤔 

1. Setter `identifier` for toppnivå
2. Looper gjennom alle `partOf`, og setter `identifier` for dem
3. Alle `partOf` looper gjennom `hasPart`, som vil føre til steg 1. igjen

Vil ikke 1-3 kjøre uendelig her, eller misforstår jeg noe?

Reply:
Responsen fra api'et baserer seg ikke på referanser, men heller duplikater av objekter, og de evige loopene blir kortsluttet etter parent -> child -> parent. Men kan være greit å sette inn en kortslutning uansett da:

```suggestion
function setIdentifierFromId(org: Organization) {
  if (org.identifier) return;
  org.identifier = getIdentifierFromId(org.id);
  org.hasPart?.map((subOrg) => setIdentifierFromId(subOrg));
  org.partOf?.map((subOrg) => setIdentifierFromId(subOrg));
}
```